pkg/agent/datamodel/sig_config.go

```diff
@@ -896,3 +923,112 @@ func withSubscription(subscriptionID string) SigImageConfigOpt {
 		c.SubscriptionID = subscriptionID
 	}
 }
+
+type Manifest struct {
```

Comment:
I'd also consider creating a separate `cache` package as well

Reply:
In my one pager. Will Daly left the comment 

I know we've talked before about having the CSE fallback to whatever is cached on the VHD, which has its own drawbacks. But I feel like it's a better user experience for nodes to continue running outdated versions of components, rather than trying to block every case where the version might be out-of-date.

so if we want to consider doing this, processing the json into structs will be more useful. Defaulting to a not desired, but cache version would probably make the user experience better. But it could leave us open to some ambiguity of customers thinking they are using one version when in actuality the are getting another